lexibank_chaconbaniwa.py

```diff
@@ -2,55 +2,77 @@
 import lingpy
 from clldutils.misc import slug
 from clldutils.path import Path
-from pylexibank.dataset import Concept
+from pylexibank import Concept
 from pylexibank.dataset import Dataset as BaseDataset
-from pylexibank.util import pb
+from pylexibank.util import progressbar
 
 
 @attr.s
-class BDConcept(Concept):
+class CustomConcept(Concept):
     Portuguese_Gloss = attr.ib(default=None)
 
 
 class Dataset(BaseDataset):
     dir = Path(__file__).parent
     id = "chaconbaniwa"
-    concept_class = BDConcept
+    concept_class = CustomConcept
 
-    def cmd_download(self, **kw):
-        pass
+    def cmd_makecldf(self, args):
+        # add sources
+        args.writer.add_sources()
 
-    def cmd_install(self, **kw):
+        # add languages
+        languages = args.writer.add_languages(lookup_factory="Name")
 
-        wl = lingpy.Wordlist(self.raw.posix("Bruzzi_Granadillo.txt"))
+        # Hard-coded fixes to segment errors in raw source
+        segments = {
+            "aʰ": "a h",
+            "ɐ̃ʰ": "ɐ̃ h",
+            "iʰ": "i h",
+            "ka": "k a",
+            "nⁱ": "n i",
+            "teː": "t eː",
+        }
 
-        with self.cldf as ds:
-            ds.add_sources(*self.raw.read_bib())
-            for k in pb(wl, desc="wl-to-cldf"):
-                ds.add_language(
-                    ID=slug(wl[k, "doculect"]), Name=wl[k, "doculect"], Glottocode="bani1255"
+        # read wordlist with lingpy
+        wl_file = self.raw_dir / "Bruzzi_Granadillo.txt"
+        wl = lingpy.Wordlist(wl_file.as_posix())
+
+        # iterate over wordlist
+        concepts = {}
+        for idx in progressbar(wl, desc="makecldf"):
+            # Concepts need to be added one by one, due to the source struct
+            # Given that the source has mixed items and no internal index,
+            # we need to build the first item of `concept_cldf_id` manually
+            if wl[idx, "concept"] not in concepts:
+                concept_cldf_id = "%i_%s" % (
+                    len(concepts) + 1,
+                    slug(wl[idx, "concept"]),
                 )
 
-                ds.add_concept(
-                    ID=slug(wl[k, "concept"]),
-                    Name=wl[k, "concept"],
-                    Concepticon_ID=wl[k, "concepticon_id"] or "",
-                    Portuguese_Gloss=wl[k, "concept_portuguese"],
+                args.writer.add_concept(
```

Comment:
While this is in principle fine, to add concepts in this way, we prefer to have a conceptlist.tsv, to start, and also the same submitted to concepticon. So we need at least an issue here that the concept list needs to be added. But given that you have them there in the wordlist file, it is easier, to already make this explicit. 

Once the conceptlist is added in etc/concepts.tsv, it can be easily added to concepticon. For this, I suggest to add an isssue and assigning Christoph (and Kristina) to it, so they can coordinate when this is added. As long as there's no official release, the conceptlist.tsv needs to remain in `etc`.

Reply:
Ok, I'll make a conceptlist in `etc`. It is good because I found at least one concept which is not mapped ("he").